nidmfsl/fsl_exporter/fsl_exporter.py

```diff
@@ -428,29 +429,129 @@ def _find_inferences(self):
+                excset_img = nib.load(filename)
+                # Get cluster connectivity
+                # There is not table display listing peaks and clusters for
+                # voxelwise correction
+                feat_post_log_file = os.path.join(
+                    analysis_dir, 'logs', 'feat4_post')
+                if os.path.isfile(feat_post_log_file):
+                    with open(feat_post_log_file, 'r') as log:
+                        feat_post_log = log.read()
+                    connectivity = self._get_connectivity(feat_post_log)
                 else:
                     warnings.warn(
-                        "'cluster' command (from FSL) not found, " +
-                        "cluster labels maps will not be exported")
-                    clust_map = None
+                        "Log file feat4_post not found, " +
+                        "connectivity information will not be reported")
+                    feat_post_log = None
+                    connectivity = 26  # FSL's default
+
+                if connectivity == 6:
+                    structure = np.array([[[0, 0, 0],
+                                           [0, 1, 0],
+                                           [0, 0, 0]],
+                                          [[0, 1, 0],
+                                           [1, 1, 1],
+                                           [0, 1, 0]],
+                                          [[0, 0, 0],
+                                           [0, 1, 0],
+                                           [0, 0, 0]]], dtype='uint8')
+                elif connectivity == 18:
+                    structure = np.array([[[0, 1, 0],
+                                           [1, 1, 1],
+                                           [0, 1, 0]],
+                                         [[1, 1, 1],
+                                          [1, 1, 1],
+                                          [1, 1, 1]],
+                                         [[0, 1, 0],
+                                          [1, 1, 1],
+                                          [0, 1, 0]]], dtype='uint8')
+                if connectivity == 26:
+                    structure = np.array([[[1, 1, 1],
+                                           [1, 1, 1],
+                                           [1, 1, 1]],
+                                          [[1, 1, 1],
+                                           [1, 1, 1],
+                                           [1, 1, 1]],
+                                          [[1, 1, 1],
+                                           [1, 1, 1],
+                                           [1, 1, 1]]], dtype='uint8')
+                else:
+                    raise Exception('Unknown connectivity: ' +
+                                    str(connectivity))
+
+                # Compute connected clusters from excursion set
+                labels, num_feat = scipy.ndimage.label(excset_img.get_data(),
```

Comment:
One last suggestion - When reading this I thought `num_feat` had something to do with the feat report but looking at the documentation for `scipy.ndimage.label` I realize this is short for number of features. As this variable is only used twice perhaps the slightly longer name `num_features` would be okay/maybe better to avoid confusion? This is only a suggestion.

Reply:
Good point! Changed to `num_labels` in c1e25d4528375ca9b5e3932b05c077b4a7f3dce1.